common/serviceerror/retry_replication.go

```diff
@@ -94,6 +94,16 @@ func (e *RetryReplication) Status() *status.Status {
 	return st
 }
 
+func (e *RetryReplication) Equal(err *RetryReplication) bool {
```

Comment:
If you'd like to be lazy here (and resilient in the face of new fields) you could use `temporalproto.Equal` from `go.temporal.io/api/temporalproto`